boofuzz/blocks/request.py

```diff
@@ -147,21 +147,23 @@ def walk(self, stack=None):
                 yield item
 
     def resolve_name(self, context_path, name):
+        if name in self.names:
+            return self.names[name]
+
+        if ".".join([context_path, name]) in self.names:
+            return self.names[".".join([context_path, name])]
+
         context_components = context_path.split(".")
         name_components = name.split(".")
+        components = context_components + name_components
 
-        if name_components[0] == "_parent":  # if path is relative
-            del name_components[0]
-            components = context_components + name_components
-            i = 0
-            while i + 1 < len(components):
-                if components[i + 1] == "_parent":
-                    del components[i + 1]
-                    del components[i]
-                else:
-                    i += 1
-        else:
-            components = context_components + name_components
+        i = 0
+        while i + 1 < len(components):
+            if components[i + 1] == "_parent":
+                del components[i + 1]
+                del components[i]
+            else:
+                i += 1
```

Comment:
Ok makes a lot more sense now, thanks for explaining @jtpereyda .

When I think about it, `_parent` should indeed be the current level, not the one above as we have it now.
So that `_parent` equals `.` and `_parent._parent` equals `..`.
On the other hand, this doesn't feel straight forward. We need some explanation for it in the documentation. I think comparing it to the Unix way makes it clear.

About two special names, I think we should stick with just one name to keep things simple. But that's just my opinion, once you get behind it, both make sense.

So we'll have to revert parts of this PR right?

Reply:
@SR4ven Yes, we would revert this part of the PR. To compensate, we would have to explain this in documentation and/or examples (separate PR of course).

Here's another one for discussion: Change the delimiter from `.` to `/` and then implement `.` and `..` as with Unix paths. It switches the analogy from objects (dot notation) to paths (slash notation). It could be more intuitive but makes me vaguely nervous. Any thoughts?

Reply:
> When I think about it, `_parent` should indeed be the current level, not the one above as we have it now.
> So that `_parent` equals `.` and `_parent._parent` equals `..`.

That doesn't make sense to me.

If `_parent` is `.`, then `_parent._parent` would be `./.` which is equal to `.`

But yeah, I get now what you were trying to achieve with the code.

Reply:
@jtpereyda switching to paths would make this all a hell lot more intuitive.

@erik-smit does have a point with `_parent._parent` equalling `.` and the makes things even more counter-intuitive. If we want to keep this analogy we'd have to use two special names like `_parent` for `..` and `_this` for `.` as @jtpereyda suggested before.

Is there any downside of switching to slash notation?

Reply:
An upside or at least neutral point is that Python already has libraries to support this kind of path parsing, including `.` and `..`. See https://stackoverflow.com/a/16595356/461834. Example:
```
>>> import posixpath as p
>>> p.normpath("../../A/B/../../C/./D/.").split('/')
['..', '..', 'C', 'D']
>>> p.normpath("../../A/B/../../C/./D/./").split('/')
['..', '..', 'C', 'D']
```
This is pretty close to our use case.

The only downside I can think of off hand is that, personally, I think the dots look a little better as a test case name than slashes would. :shrug:

Reply:
True, the dots look better.
I'm not sure which notation is the better choice. The dot notation is prettier but less intuitive for relative paths, the slash notation it's the other way around.

On the other hand, how frequently will the relative paths be used? Feels like this is rarely used feature. Maybe a little less intuitive usage with proper documentation is acceptable here.

Reply:
OK, let's go with the slashes and dots and document. I will do this if I can squeeze it in soon. Would like to get this all merged.